### PR TITLE
chore: Setup UA actions for Auth & Storage

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

Adds user agent actions for Auth & Storage APIs. User agent information is already present for Analytics & Notifications.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/providers/cognito/apis/resetPassword.ts [hunk 1]
```diff
@@ -34,7 +35,10 @@ export async function resetPassword(
 	assertTokenProviderConfig(authConfig);
 	const clientMetadata = input.options?.serviceOptions?.clientMetadata;
 	const res = await forgotPassword(
-		{ region: getRegion(authConfig.userPoolId) },
+		{ 
+			region: getRegion(authConfig.userPoolId),
+			userAgentValue: getAuthUserAgentValue(AuthAction.ResetPassword)
```
Comment (elorzafe) [inline packages/auth/src/providers/cognito/apis/resetPassword.ts:40]: nit: why ResetPassword and no forgotPassword

File: packages/storage/src/providers/s3/apis/internal/getProperties.ts [hunk 2]
```diff
@@ -2,24 +2,33 @@
...
 	);
 
-	const response = await headObject(s3Config, {
-		Bucket: bucket,
-		Key: `${keyPrefix}${key}`,
-	});
+	const response = await headObject(
+		{
+			...s3Config, 
+			userAgentValue: getStorageUserAgentValue(action || StorageAction.GetProperties)
```
Comment (elorzafe) [inline packages/storage/src/providers/s3/apis/internal/getProperties.ts:25]: nit use `??` @cshfang

File: packages/auth/src/providers/cognito/apis/resetPassword.ts (hunk 1 above)
Comment (jimblanc) [reply packages/auth/src/providers/cognito/apis/resetPassword.ts:40]: The action tracks the Amplify API that the customer invoked (in this case `resetPassword`), not the underlying Cognito API.

**Review by elorzafe — COMMENTED** [review]
Is it possible to check if the headers `x-amz-user-agent` is present?

**Comment by jimblanc** [comment]
> Is it possible to check if the headers `x-amz-user-agent` is present?

The tests for the custom clients already do that, I'm just making sure we're passing the correct value to the clients.

**Review by elorzafe — DISMISSED** [review]

**Review by AllanZhengYP — DISMISSED** [review]
Looks good to me🫡

**Review by kvramyasri7 — DISMISSED** [review]
Storage changes look good to me

**Review by kvramyasri7 — APPROVED** [review]
🚢

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**